starter-code/src/functions-and-arrays.js

```diff
@@ -1,100 +1,457 @@
 // Find the maximum
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else {
+    return num2;
+  }
+}
+maxOfTwoNumbers(6, 7);
+maxOfTwoNumbers(1, 9);
+maxOfTwoNumbers(67, 7);
 
 // Finding Longest Word
 var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
 ];
 
+function findLongestWord(array) {
+  var palabraLarga = "";
+
+  if (array.length >= 1) {
+    array.forEach(function(value) {
+      if (value.length > palabraLarga.length) {
+        palabraLarga = value;
+      }
+    });
+    console.log(palabraLarga);
+    return palabraLarga;
+  }
+}
+
+findLongestWord(words);
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(array) {
+  var suma = 0;
+  for (var i = 0; i < array.length; i++) {
+    suma += array[i];
+  }
+  return suma;
+}
+
+sumArray();
 // Calculate the Average
 
+function averageNumbers(array) {
+  var suma = 0;
+  if (array.length >= 1) {
+    for (var i = 0; i < array.length; i++) {
+      suma += array[i];
+    }
+    return suma / array.length;
+  }
+  
+}
+
+averageNumbers(numbersAvg);
+
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
 // Array of Strings
+
+function averageWordLength(array){
+  var suma = 0;
+  if (array.length >= 1) {
+    for (var i = 0; i < array.length; i++) {
+      suma += array[i].length;
+    }
+    return suma / array.length;
+  }
+} 
+
 var wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+averageWordLength(wordsArr);
+
 // Unique Arrays
 var wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
+function uniquifyArray(array){
+
+  for(var i = 0;i < array.length; i++ ){
+    var cadena = array[i];
+    for(var j=0; j < array.length; j++){
+      var arrayFinal;
+      if(cadena === array[i] && !array.indexOf(array[i])){
```

Comment:
No estais usando el segundo bucle for , ya que las comprobaciones las estais haciendo siempre con la 'i'.